File: branches/Medusa/Source/Fantasy.Jobs/StateBagItem.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using Fantasy.XSerialization;
using System.Xml;
using System.ComponentModel;
using System.Collections;
using System.Xml.Linq;

namespace Fantasy.Jobs
{
    
    [XSerializable("state", NamespaceUri= Consts.XNamespaceURI)] 
    public class StateBagItem : IXSerializable 
    {
       
        public string Name { get; set; }

        private object _value;

        public object Value
        {
            get { return _value; }
            set { _value = value; }
        }


        #region IXSerializable Members

        void IXSerializable.Load(IServiceProvider context, XElement element)
        {
            this.Name = element.Name.LocalName;
            XNamespace ns = Consts.XNamespaceURI;
            if(! String.IsNullOrEmpty((string)element.Attribute(ns + "type")))
            {
                
                Type t = Type.GetType((string)element.Attribute(ns + "type"), true);
                if (t.IsDefined(typeof(XSerializableAttribute), false))
                {
                   
                    XSerializer ser = new XSerializer(t);
                    this.Value = ser.Deserialize(element);
                }
                else
                {
                    TypeConverter convert = XHelper.Default.CreateXConverter(t);
                    this.Value = convert.ConvertTo(element.Value, t); 
                }
            }

        }

        void IXSerializable.Save(IServiceProvider context, XElement element)
        {
 
            if (Value != null)
            {
                XNamespace ns = Consts.XNamespaceURI;
                Type t = Value.GetType();
                element.SetAttributeValue(ns + "type", string.Format("{0}, {1}" , t.FullName, t.Assembly.GetName().Name) );

                if (t.IsDefined(typeof(XSerializableAttribute), false))
                {
                    XSerializer ser = new XSerializer(t);
                    ser.Serialize(element, Value);
                }
                else
                {
                    TypeConverter convert = XHelper.Default.CreateXConverter(t);
                    string s = (string)convert.ConvertFrom(this.Value);
                    element.Value = s;
                }


            }
        }

        #endregion
    }

    public class StateBagItemsSerializer : IXCollectionSerializer
    {
        public void Save(IServiceProvider context, XElement element, IEnumerable collection)
        {
          
            foreach (StateBagItem item in collection)
            {
                XElement childElement = new XElement(element.Name.Namespace + item.Name);
                element.Add(childElement);
                ((IXSerializable)item).Save(context, childElement);
            }
        }

        public IEnumerable Load(IServiceProvider context, XElement element)
        {
            ArrayList rs = new ArrayList();
            foreach (XElement childElement in element.Elements())
            {
                StateBagItem item = new StateBagItem();
                ((IXSerializable)item).Load(context, childElement);
                rs.Add(item);
            }

            return rs;
        }
    }

}

File: AutoConverted/Fantasy.Jobs/StateBagItem.java
package Fantasy.Jobs;

import Fantasy.XSerialization.*;

//C# TO JAVA CONVERTER TODO TASK: Java annotations will not correspond to .NET attributes:
//[XSerializable("state", NamespaceUri= Consts.XNamespaceURI)]
public class StateBagItem extends IXSerializable
{

	private String privateName;
	public final String getName()
	{
		return privateName;
	}
	public final void setName(String value)
	{
		privateName = value;
	}

	private Object _value;

	public final Object getValue()
	{
		return _value;
	}
	public final void setValue(Object value)
	{
		_value = value;
	}


//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
		///#region IXSerializable Members

	private void Load(IServiceProvider context, XElement element)
	{
		this.setName(element.getName().LocalName);
		XNamespace ns = Consts.XNamespaceURI;
		if(! DotNetToJavaStringHelper.isNullOrEmpty((String)element.Attribute(ns + "type")))
		{

			java.lang.Class t = java.lang.Class.forName((String)element.Attribute(ns + "type"), true);
			if (t.IsDefined(XSerializableAttribute.class, false))
			{

				XSerializer ser = new XSerializer(t);
				this.setValue(ser.Deserialize(element));
			}
			else
			{
				TypeConverter convert = XHelper.Default.CreateXConverter(t);
				this.setValue(convert.ConvertTo(element.getValue(), t));
			}
		}

	}

	private void Save(IServiceProvider context, XElement element)
	{

		if (getValue() != null)
		{
			XNamespace ns = Consts.XNamespaceURI;
			java.lang.Class t = getValue().getClass();
			element.SetAttributeValue(ns + "type", String.format("%1$s, %2$s", t.FullName, t.Assembly.GetName().getName()));

			if (t.IsDefined(XSerializableAttribute.class, false))
			{
				XSerializer ser = new XSerializer(t);
				ser.Serialize(element, getValue());
			}
			else
			{
				TypeConverter convert = XHelper.Default.CreateXConverter(t);
				String s = (String)convert.ConvertFrom(this.getValue());
				element.setValue(s);
			}


		}
	}

//C# TO JAVA CONVERTER TODO TASK: There is no preprocessor in Java:
		///#endregion
}